devbox.json

```diff
@@ -2,23 +2,23 @@
   "packages": [
     "bash@latest",
     "coreutils@latest",
-    "clusterctl@latest",
-    "envsubst@latest",
+    "clusterctl@1.8.0",
+    "envsubst@1.4.2",
     "fping@latest",
-    "gnumake@latest",
-    "ginkgo@latest",
-    "go@latest",
-    "golangci-lint@latest",
-    "gotestsum@latest",
+    "gnumake@4.4.1",
+    "ginkgo@2.19.1",
+    "go@1.22.5",
+    "golangci-lint@1.62.2",
+    "gotestsum@1.11.0",
     "kubernetes-helm@latest",
     "kind@latest",
     "ko@latest",
     "kubectl@latest",
-    "kubernetes-code-generator@latest",
-    "kubernetes-controller-tools@latest",
-    "kustomize@latest",
-    "mockgen@latest",
-    "yamllint@latest",
+    "kubernetes-code-generator@0.25.4",
+    "kubernetes-controller-tools@0.14.0",
+    "kustomize@5.4.3",
+    "mockgen@0.4.0",
+    "yamllint@1.35.1",
```

Comment:
Do we need to pin all these versions?

I know we had trouble with newer versions of golangci-lint.

Reply:
yes, we need

Reply:
Why do we need this? Can we desperate this out to a new branch?

Reply:
@faiq  Couple of latest linters contains some bugs which extensively using the memory. With this bugs `make lint` was killed by OOM killer. We met these bugs in this PR and explore it with @dlipovetsky. Fixing golang version and tools versions help to fix linters issues.

Reply:
I think we need to think through this part of the PR more carefully. For example, why do we pin go v1.22.5, when the go module specifies a different version:

```
module github.com/nutanix-cloud-native/cluster-api-provider-nutanix

go 1.22.1

toolchain go1.22.2
```

I think it would be very helpful to pull this change out into its own PR, to be honest.

And I would like @thunderboltsid to provide his input, too.

Reply:
@dlipovetsky All these version was taken from actual version of `devbox.lock`. E.g. golang: https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/blob/main/devbox.lock#L383
Without these changes in this PR we meet linters issues.

jfyi: @thunderboltsid

Reply:
discussed this offline with Ilya. Let's try to keep out any devbox related changes from this PR. The go module version is largely for anyone importing capx as a dependency (usually for API package) and doesn't have to correlate with the actual go version we use for compiling our programs (as long as that version is greater than the module prescribed version).